doc/developer/design/20231110_aws_connections.md

```diff
@@ -0,0 +1,164 @@
+handle the AssumeRole scenario.
+
+In the parser we already support `ROLE ARN` option which can be renamed to `ASSUME ROLE ARN`
+to better reflect the usage. We will need to add an `ASSUME ROLE SESSION NAME` option as well.
+
+The AWS Connection create statement with all the options will now look like:
+```sql
+CREATE CONNECTION <connector_name> TO AWS (
+  ACCESS KEY ID [[=] <value>], -- existing
+  SECRET ACCESS KEY [[=] SECRET <value>], -- existing
+  TOKEN [[=] SECRET <value>], -- existing
+  ASSUME ROLE ARN [[=] <value>], -- rename existing from ROLE ARN
+  ASSUME ROLE SESSION NAME [[=] <value>], -- new option
+  ENDPOINT [[=] <value>], -- existing
+  REGION [[=] <value>] -- existing
+);
+```
+
+Users should be able to provide either the `ACCESS KEY` options or the `ASSUME ROLE` options,
+but not both.
+
+#### External ID and principal
+We already have external ID prefix provided in the catalog state.
+The connection ID will be appended to the external ID prefix to get the complete
+External ID and then stored in the new `mz_aws_connections` table for that connection ID.
+
+For principal, we'll create one global role which all environmentd/clusterd-s can assume.
+This will add some good restrictions on who could assume the customers role and also
+allow for migration between environment clusters.
+
+Note: There's a hard limit of 1 hour on the session duration when [Role chaining in AWS](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_terms-and-concepts.html).
+This can be revisited for workarounds if we hit this issue.
+
+Other alternative solutions were:
+1. Using our root account `arn:aws:iam::<account_id>:root`. This would work but this is not
+secure enough since any user/role in this account can assume the customer's role.
+2. An IAM user `arn:aws:iam::<account_id>:user/<id>` created for the customer's environment similar
+to [Snowflake](https://docs.snowflake.com/en/user-guide/data-load-s3-config-storage-integration#step-4-retrieve-the-aws-iam-user-for-your-snowflake-account).
+This could work if we keep them in a secret and rotate regularly, this would give
+each customer a unique principal. But we'd have to rotate very frequently and we may also
+need to coordinate that role across all users of the credentials which seems like a lot of overhead.
+3. An IAM role `arn:aws:iam::<account_id>:role/<id>` we create for each customer environment. This
+will not scale, there's a maximum limit of [5000 roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entities)
+we can create per account.
+
+#### `VALIDATE CONNECTION`
+We can make use of the [GetCallerIdentity AWS API](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html)
+to check if the credentials are valid. We should make the call with the given External ID and
+without to make sure that the user has a secure setup. If we are able to access without the External
+ID, we should treat that as an error.
```

Comment:
Wow, validating that we do _not_ have access without the external ID is a great idea!

Reply:
This was @jubrad's suggestion :)